### PR TITLE
QRCodeScanner: import QR code images from gallery

### PR DESCRIPTION
**Pull request by shubhamkmr04**
# Description

Relates to issue: **https://github.com/ZeusLN/zeus/issues/833**

The description of the Issue and how did I approach it with the solution is in this [documentation](https://docs.google.com/document/d/1aSSd3oH0Ud_W7hNUZrbk1uBGpWxGRQDKnpb58-hjQBg/edit)

This pull request is categorized as a:

- [X] New feature
- [ ] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance 
- [ ] Other

## Checklist
- [X] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [X] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [X] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [X] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [ ] No, I’m a fool
- [ ] Yes
- [X] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [X] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [X] LND (REST)
- [ ] LND (Lightning Node Connect)
- [ ] Core Lightning (c-lightning-REST)
- [ ] Core Lightning (Spark)
- [ ] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [X] Contributors will need to run `yarn` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: backends/LND.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ export default class LND {
         route: string,
         ws?: boolean
     ) => {
-        const hostPath = host.includes('://') ? host : `https://${host}`;
+        const hostPath = host.includes('://') ? host : `http://${host}`;
```
Comment (kaloudis) [inline backends/LND.ts:147]: gotta undo this change as it's only for dev

File: package.json [hunk 2]
```diff
@@ -75,6 +76,7 @@
     "react-native-encrypted-storage": "4.0.2",
     "react-native-gesture-handler": "2.2.0",
     "react-native-hce": "0.1.2",
+    "react-native-image-picker": "^5.0.1",
```
Comment (kaloudis) [inline package.json:79]: I like to lock the version numbers here. Please remove the `^` from both new additions and run yarn again, please.

File: components/QRCodeScanner.tsx [hunk 3]
```diff
@@ -67,6 +70,29 @@ export default class QRCodeScanner extends React.Component<QRProps, QRState> {
...
     };
 
+    handleOpenGallery = () => {
+        launchImageLibrary(
+            {
+                mediaType: 'photo'
+            },
+            (response) => {
+                if (response.didCancel) {
+                    console.log('USER CANCELLED IT');
```
Comment (kaloudis) [inline components/QRCodeScanner.tsx:80]: Do we need to take any other action in this instance? perhaps we remove this block and wrap the other action in `if (!response.didCancel)`

File: backends/LND.ts (hunk 1 above)
Comment (shubhamkmr04) [reply backends/LND.ts:147]: got it!

File: components/QRCodeScanner.tsx (hunk 3 above)
Comment (shubhamkmr04) [reply components/QRCodeScanner.tsx:80]: Okay sure. We can do this for now. Though we might use this instance to change a loading variable like `loading == false` in future

File: android/app/build.gradle [hunk 4]
```diff
@@ -202,11 +202,12 @@ dependencies {
     implementation 'com.android.support:multidex:1.0.3'
     implementation project(':react-native-randombytes')
     implementation "androidx.core:core:1.1.0"
-    implementation "com.facebook.react:react-native:+"  // From node_modules
+    implementation "com.facebook.react:react-native:0.70.6!!"  // From node_modules
```
Comment (kaloudis) [inline android/app/build.gradle:205]: is it required to specify the version here?

Comment (shubhamkmr04) [reply android/app/build.gradle:205]: Yes I was getting some BUILD error, so I had to specify the version. Like I read it from [here](https://stackoverflow.com/questions/74325460/a-failure-occurred-while-executing-com-android-build-gradle-internal-tasks-merge)

**Comment by shubhamkmr04** [comment]
The previous commit introduce a patch file to upgrade build.gradle file of `@remobile/react-native-qrcode-local-image`

File: package.json [hunk 5]
```diff
@@ -11,13 +11,14 @@
   },
   "scripts": {
     "start": "react-native start",
-    "postinstall": "rn-nodeify --install --hack; npx jetify",
+    "postinstall": "rn-nodeify --install --hack; npx jetify; git apply patches/rnqli-build.gradle.patch",
```
Comment (kaloudis) [inline package.json:14]: up here let's turn the patch func to `yarn run patch` since we have it defined below

Comment (shubhamkmr04) [reply package.json:14]: Okay! then would I have to shift "postinstall" below "patch" ?

Comment (kaloudis) [reply package.json:14]: I don't think you have to but please double check before committing the change

Comment (shubhamkmr04) [reply package.json]: tested that. Its working perfectly

**Review by kaloudis — APPROVED** [review]
tACK. great job!
